packages/fireblocks-signing-manager/src/lib/fireblocks/fireblocks.spec.ts

```diff
@@ -0,0 +1,128 @@
+
+      expect(result).toEqual([]);
+    });
+
+    it('should return derived keys', async () => {
+      const publicKeySpy = jest.spyOn(fireblocks.fireblocksSdk, 'getPublicKeyInfo');
+
+      publicKeySpy.mockResolvedValue(key1);
+      await fireblocks.deriveKey(key1.derivationPath);
+
+      publicKeySpy.mockResolvedValue(key2);
+      await fireblocks.deriveKey(key2.derivationPath);
+
+      const result = fireblocks.fetchDerivedKeys();
+
+      expect(result).toEqual(expect.arrayContaining([key1, key2]));
+
+      publicKeySpy.mockRestore();
+    });
+  });
+
+  describe('method: deriveKey', () => {
+    it('should return the key', async () => {
+      const publicKeySpy = jest.spyOn(fireblocks.fireblocksSdk, 'getPublicKeyInfo');
+
+      publicKeySpy.mockResolvedValue(key1);
+      const key = await fireblocks.deriveKey(key1.derivationPath);
+
+      expect(key).toEqual(key1);
+    });
+  });
+
+  describe('method: signData', () => {
+    it('should return signed data', async () => {
+      const publicKeySpy = jest.spyOn(fireblocks.fireblocksSdk, 'getPublicKeyInfo');
+      publicKeySpy.mockResolvedValue(defaultKey);
+
+      const createTxSpy = jest.spyOn(fireblocks.fireblocksSdk, 'createTransaction');
+
+      createTxSpy.mockResolvedValue({ id: '1', status: TransactionStatus.PENDING_AUTHORIZATION });
+
+      const getTxSpy = jest.spyOn(fireblocks.fireblocksSdk, 'getTransactionById');
+
+      const expectedSignature = '123';
+      getTxSpy.mockResolvedValueOnce({
+        id: '1',
+        status: TransactionStatus.PENDING_SIGNATURE,
+      } as TransactionResponse);
+      getTxSpy.mockResolvedValueOnce({
+        id: '1',
+        status: TransactionStatus.COMPLETED,
+        signedMessages: [{ signature: { fullSig: expectedSignature } }],
+      } as TransactionResponse);
+
+      const result = await fireblocks.signData(defaultKey.derivationPath, 'test data', 'test note');
+
+      expect(getTxSpy).toHaveBeenCalledTimes(2);
+      expect(result).toBe(expectedSignature);
+
+      publicKeySpy.mockRestore();
+      createTxSpy.mockRestore();
+      getTxSpy.mockRestore();
+    });
+
+    it('should throw an error if a transaction is rejected', async () => {
+      const createTxSpy = jest.spyOn(fireblocks.fireblocksSdk, 'createTransaction');
+      createTxSpy.mockResolvedValue({ id: '1', status: TransactionStatus.PENDING_AUTHORIZATION });
+
+      const getTxSpy = jest.spyOn(fireblocks.fireblocksSdk, 'getTransactionById');
+      getTxSpy.mockResolvedValue({
+        id: '1',
+        status: TransactionStatus.REJECTED,
+      } as TransactionResponse);
+
+      const expectedError = new NoTransactionSignature(
+        'Fireblocks transaction response with status: "REJECTED" had no signature'
+      );
+
+      await expect(
+        fireblocks.signData(defaultKey.derivationPath, 'test data', 'test note')
+      ).rejects.toThrowError(expectedError);
+
+      createTxSpy.mockRestore();
```

Comment:
You forgot to restore getTxSpy. I think we can add `restoreMocks: true` in jest.config similar to what you suggested in REST API, here as well

Reply:
I set the value for all the signing managers here just so we have consistent behavior.